Sources/StreamChat/APIClient/RequestEncoder.swift

```diff
@@ -151,10 +151,9 @@ struct DefaultRequestEncoder: RequestEncoder {
                     updatedRequest.setHTTPHeaders(.jwtStreamAuth, .authorization(token.rawValue))
                 }
                 completion(.success(updatedRequest))
-            case .failure(_ as ClientError.WaiterTimeout):
-                // We complete with a success to account for the most probable case for the timeout: No connection.
-                // That way, when reaching the APIClient, we would properly report a connection error.
-                completion(.success(request))
+            case let .failure(error as ClientError.WaiterTimeout):
+                // The APIClient will treat a waiter timeout differently than the other ones
+                completion(.failure(error))
```

Comment:
Why are we treating it differently?

Reply:
We are treating them differently in the sense that we are no masking the error, but returning the original one so that it can be properly handled by the receiver